app/scripts/modules/core/pipeline/config/triggers/webhook/webhookTrigger.module.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+let angular = require('angular');
+
+module.exports = angular.module('spinnaker.core.pipeline.trigger.webhook', [
+    require('../../../../../core/config/settings.js'),
+  ])
+  .config(function (pipelineConfigProvider) {
+    pipelineConfigProvider.registerTrigger({
+      label: 'Webhook',
+      description: 'Executes the pipeline on a Webhook POST',
+      key: 'webhook',
+      controller: 'WebhookTriggerCtrl as ctrl',
+      controllerAs: 'vm',
+      templateUrl: require('./webhookTrigger.html'),
+      popoverLabelUrl: require('./webhookPopoverLabel.html'),
+    });
+  })
+  .controller('WebhookTriggerCtrl', function (trigger, $scope) {
```

Comment:
This was so promising. :-)  However, it appears that a null ```{source}``` means that the URL is not found in either gate or echo.
My previous comment about it working with ```http://<my_echo_address>/webhooks/type/``` must've been because I had not removed my new ```/webhooks/type``` handler from the  WebhooksController.groovy.

I think my next step is add the ```/webhooks/type``` into Gate and raise a PR for that.